### PR TITLE
Use upstream implementation as the base, maintain only diffs

### PR DESCRIPTION
**Pull request by fabianvf**

### CODE REVIEW

File: openshift/dynamic/discovery.py [hunk 1]
```diff
@@ -1,22 +1,31 @@
...
+from kubernetes.dynamic.discovery import (  # noqa
+    CacheEncoder,
+    CacheDecoder,
+    DISCOVERY_PREFIX,
+    ResourceGroup,
+)
+from kubernetes.dynamic.exceptions import (
+    ResourceNotFoundError,
+    ResourceNotUniqueError,
+    ApiException,
```
Comment (asmacdo) [inline openshift/dynamic/discovery.py:21]: Optional Nit: For pep8 these should be in alphabetical order, but not a big deal. If the need to be imported in this order, a comment would be nice.

File: tox.ini [hunk 2]
```diff
@@ -1,14 +1,14 @@
 [tox]
 envlist =
-    py{27,35}-{lint}
-    py{27,35}-openshift-unit
+    py{27,39}-{lint}
```
Comment (asmacdo) [inline tox.ini:3]: Looks like we dont want 27 here anymore.

**Review by asmacdo — APPROVED** [review]
/lgtm

**Review by jmrodri — APPROVED** [review]
/lgtm
Love code removal

**Pull request merged**